tests/test_fetch_url.py

```diff
@@ -96,3 +160,47 @@ async def test_fetch_url_javascript_driven_site(fetch_url_tool: FetchURL) -> Non
     # If it fails, should indicate extraction issues
     if isinstance(result, ToolError):
         assert "failed to extract meaningful content" in result.message.lower()
+
+
+@pytest.mark.asyncio
+async def test_fetch_url_mocked_http_responses(
+    fetch_url_tool: FetchURL,
+    mock_html_server: MockServerFactory,
+) -> None:
+    """Test fetching multiple mocked HTTP responses."""
+
+    async def mocked_fetch(resp: str, *, content_type: str = "text/html") -> ToolReturnType:
+        server_url = await mock_html_server(resp, content_type=content_type)
+        return await fetch_url_tool(Params(url=f"{server_url}/"))
+
+    # plain markdown. Real example: https://lucumr.pocoo.org/2025/10/17/code.md
+    plain_markdown = """\
+# Title
+
+This is a markdown document.
+"""
```

Comment:
For this test, it will be `"failed to extract meaningful content"` before the change.